shared/src/Twin.cs

```diff
@@ -151,6 +151,19 @@ public Twin(TwinProperties twinProperties)
         [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
         public X509Thumbprint X509Thumbprint { get; internal set; }
 
+        /// <summary>
+        /// The scope of the device. Auto-generated and immutable for edge devices and modifiable in leaf devices to create child/parent relationship.
+        /// </summary>
+        [DefaultValue(null)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public string DeviceScope { get; internal set; }
+
+        /// <summary>
+        /// The scopes of the upper level edge devices if applicable. Only available for edge devices.
+        /// </summary>
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public virtual IReadOnlyList<string> ParentScopes { get; internal set; } = new List<string>();
```

Comment:
This is a ReadOnly list. Any use of assigning the default value? I guess it will either be de-serialized or not based on weather the service returns it or not right?

Reply:
Can this be set by users or is it just something the service auto-popylates?

Reply:
I think you are right. I'll remove the initialization.